crates/runtime/src/object_store_registry/mod.rs

```diff
@@ -101,7 +101,11 @@ impl SpiceObjectStoreRegistry {
                     s3_builder = s3_builder.with_skip_signature(true);
                     load_credentials_from_environment = false;
                 }
-                None => {}
+                None => {
+                    // Default to public if no auth is provided
+                    s3_builder = s3_builder.with_skip_signature(true);
+                    load_credentials_from_environment = false;
+                }
```

Comment:
I don't think we should default to public if auth isn't specified - we should ensure public works if they don't specify it, but we should attempt to use credentials if they exist.